### PR TITLE
Expand Supervisor to support loading additional scripts before user scripts

### PR DESCRIPTION
**Pull request by ckreibich**
I hit another scenario in which the fact that the Supervisor's script customization adds to the very end of the script pile tripped me up. We've touched this on occasion in the past and I've so far worked around it, but it always feels clumsy, so here's an additional vector for loading before the user-provided ones.

I'm unsure about naming, thoughts welcome. Something like `$pre_scripts` and `$post_scripts` seems nicer, but that would break existing scripts relying on the existing naming. We could add `$post_scripts` and deprecate `$scripts`.

### CODE REVIEW

**Comment by ckreibich** [comment]
I rather like `addl_base_scripts` and `addl_user_scripts` for the two vectors.

**Comment by ckreibich** [comment]
> I rather like addl_base_scripts and addl_user_scripts for the two vectors.

I went ahead and added these.

File: src/supervisor/Supervisor.h [hunk 1]
```diff
@@ -198,7 +198,19 @@ class Supervisor : public iosource::IOSource
...
+		std::vector<std::string> addl_base_scripts;
+		/**
+		 * Additional script filename/paths that the node should load
+		 * after any user-specified scripts.
+		 */
+		std::vector<std::string> addl_user_scripts;
+		/**
+		 * The former name for addl_user_scripts, now deprecated.  Use
+		 * addl_user_scripts instead. The scriptland equivalent has a
+		 * proper deprecation tag.
```
Comment (timwoj) [inline src/supervisor/Supervisor.h:213]: Any reason to not mark the C++ side deprecated too?

File: testing/btest/supervisor/config-scripts.zeek [hunk 2]
```diff
@@ -36,7 +38,15 @@ event zeek_init()
...
-		print node_output_file, "supervised node zeek_init()";
+
+@ifdef ( Pre::counter )
+		# Even though this else-block runs only in the supervised node,
+		# it still needs to parse correctly in the Supervisor, so we
+		# need to condition this with availability of the counter (which
+		# only exists in the supervised node).
+		++Pre::counter;
+		print node_output_file, fmt("supervised node zeek_init(), counter at %s", Pre::counter);
+@endif
```
Comment (timwoj) [inline testing/btest/supervisor/config-scripts.zeek:49]: Is this block related to this PR or something else?

**Review by timwoj — CHANGES_REQUESTED** [review]

File: src/supervisor/Supervisor.h (hunk 1 above)
Comment (ckreibich) [reply src/supervisor/Supervisor.h:213]: I wondered about that. It adds a bunch of warnings to our build that seemed excessive to me given the script-layer deprecation, but it would technically be correct to deprecate there as well. Happy to do as you prefer.

File: testing/btest/supervisor/config-scripts.zeek (hunk 2 above)
Comment (ckreibich) [reply testing/btest/supervisor/config-scripts.zeek:49]: This PR — the test verifies the load order by adding the counter in the post-base scripts, using it in the test script itself, and again in the post-user ones.

File: src/supervisor/Supervisor.h (hunk 1 above)
Comment (timwoj) [reply src/supervisor/Supervisor.h:213]: It's not part of a `detail` namespace so we should probably follow the process. At worst, you can add pragmas to the Zeek code to silence the warnings for our builds.

Comment (ckreibich) [reply src/supervisor/Supervisor.h]: Added, Tim. You can see the noise here: https://cirrus-ci.com/task/6385802665000960?logs=build#L3071

Comment (ckreibich) [reply src/supervisor/Supervisor.h]: I've pushed an update with pragma use as per our discussion.

**Review by timwoj — APPROVED** [review]
One nit. LGTM otherwise.

File: src/supervisor/Supervisor.h [hunk 3]
```diff
@@ -143,6 +143,16 @@ class Supervisor : public iosource::IOSource
 	 */
 	struct NodeConfig
 		{
+#pragma GCC diagnostic push
```
Comment (timwoj) [inline src/supervisor/Supervisor.h:146]: Maybe just add a comment here about why these are defined and that they can possibly be removed later.

Comment (ckreibich) [reply src/supervisor/Supervisor.h:146]: Good call Tim, update pushed.
